giskard/testing/tests/data_quality.py

```diff
@@ -0,0 +1,299 @@
+            dataset_copy[column] = le.fit_transform(dataset_copy[column])
+
+    # Prepare the data
+    x = dataset_copy[list(reference_columns)]
+    y = dataset_copy[labelled_column]
+
+    # Combine the data and labels
+    data = pd.concat([x, y], axis=1)
+
+    # Train the Isolation Forest model
+    model = IsolationForest(contamination=0.1)
+    model.fit(data)
+
+    # Predict the anomalies in the data
+    anomalies = model.predict(data) == -1
+
+    # Check if any of the anomalies have different labels
+    mislabelled_data = dataset.df[anomalies]
+
+    if not mislabelled_data.empty:
+        message = f"Mislabelled data found: \n{mislabelled_data}"
+        return TestResult(passed=False, metric_name="consistency", metric=0, messages=message)
+
+    return TestResult(passed=True, metric_name="consistency", metric=1)
+
+@test(name="Feature Importance Test")
+def feature_importance(dataset: Dataset, target_column: str, feature_columns: Iterable[str]):
+    """
+    Test for evaluating the importance of each feature to the target variable.
+
+    Args:
+        dataset (giskard.Dataset): The dataset to test.
+        target_column (str): The column containing the target variable.
+        feature_columns (Iterable[str]): The columns containing the features.
+
+    Returns:
+        TestResult: The result of the test, containing the feature importances.
+        The more value the more important a feature is with respect to target variable.
+    """
+    # Prepare the data
+    features = list(feature_columns)
+    x = dataset.df[features]
+    y = dataset.df[target_column]
+    # Train the Random Forest model
+    model = RandomForestClassifier()
+    model.fit(x, y)
+    # Get the feature importances
+    importances = model.feature_importances_
+    feature_importances = dict(zip(features, importances))
+    # Create a message containing the feature importances
+    message = f"Feature importances: \n{feature_importances}"
+
+    return TestResult(passed=True, metric_name="feature_importance",
+                      metric=importances, messages=message)
+
+@test(name="Class Imbalance Test")
+def class_imbalance(dataset: Dataset, target_column: str):
+    """
+    Test for assessing the distribution of classes in classification problems.
+
+    Args:
+        dataset (giskard.Dataset): The dataset to test.
+        target_column (str): The column containing the target variable.
+
+    Returns:
+        TestResult: The result of the test, containing the class proportions.
+    """
+    # Convert classes to strings and calculate the class proportions
+    class_counts = Counter(dataset.df[target_column].astype(str))
+    total_count = len(dataset.df)
+    class_proportions = {cls: count / total_count for cls, count in class_counts.items()}
+
+    # Create a message containing the class proportions
+    message = f"Class proportions: \n{class_proportions}"
+
+    return TestResult(passed=True,
```

Comment:
This test always return `True`, we should probably add some check to ensure we don't have over or under represented classes. 

Maybe we can introduce some thresholds